File: Go/Algorithms/insertionSort.go
package main

func insertionSort(array []int) []int {
	for i := 1; i < len(array); i++ {
		var key int = array[i];
		var j int = i - 1;

		for j >= 0 && array[j] > key {
			array[j+1] = array[j];
			j--;
		}
		array[j+1] = key;
	}
	return array;
}

File: Nim/Algorithms/insertion_sort.nim
proc insertion_sort(arr: var seq[int]): seq[int] = 
    for i in 1..<arr.len:
        var key: int = arr[i]
        var j: int = i - 1

        while j >= 0 and arr[j] > key:
            arr[j+1] = arr[j]
            j -= 1
        
        arr[j+1] = key
    
    return arr